client/blocks/stripe-utils/normalize.js

```diff
@@ -0,0 +1,176 @@
+			.slice( 0, 1 )
+			.join( ' ' ),
+		last_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 1 )
+			.join( ' ' ),
+		company: '',
+		address_1:
+			typeof shippingAddress.addressLine[ 0 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 0 ],
+		address_2:
+			typeof shippingAddress.addressLine[ 1 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 1 ],
+		city: shippingAddress.city,
+		state: shippingAddress.region,
+		country: shippingAddress.country,
+		postcode: shippingAddress.postalCode.replace( ' ', '' ),
+	};
+	return address;
+};
+
+/**
+ * Normalizes shipping option shape selection from Stripe's shipping option
+ * object to the expected shape for cart shipping option selections.
+ *
+ * @param {StripeShippingOption} shippingOption The customer's selected shipping
+ *                                              option.
+ *
+ * @return {string[]}  An array of ids (in this case will just be one)
+ */
+const normalizeShippingOptionSelectionsForCheckout = ( shippingOption ) => {
+	return shippingOption.id;
+};
+
+/**
+ * Returns the billing data extracted from the stripe payment response to the
+ * CartBillingData shape.
+ *
+ * @param {StripePaymentResponse} paymentResponse Stripe's payment response
+ *                                                object.
+ *
+ * @return {CartBillingAddress} The cart billing data
+ */
+const getBillingData = ( paymentResponse ) => {
+	const source = paymentResponse.source;
+	const name = source && source.owner.name;
+	const billing = source && source.owner.address;
+	const payerEmail = paymentResponse.payerEmail || '';
+	const payerPhone = paymentResponse.payerPhone || '';
+	return {
+		first_name: name ? name.split( ' ' ).slice( 0, 1 ).join( ' ' ) : '',
+		last_name: name ? name.split( ' ' ).slice( 1 ).join( ' ' ) : '',
+		email: ( source && source.owner.email ) || payerEmail,
+		phone:
+			( source && source.owner.phone ) ||
+			payerPhone.replace( '/[() -]/g', '' ),
+		country: ( billing && billing.country ) || '',
+		address_1: ( billing && billing.line1 ) || '',
+		address_2: ( billing && billing.line2 ) || '',
+		city: ( billing && billing.city ) || '',
+		state: ( billing && billing.state ) || '',
+		postcode: ( billing && billing.postal_code ) || '',
+		company: '',
+	};
+};
+
+/**
+ * This returns extra payment method data to add to the payment method update
+ * request made by the checkout processor.
+ *
+ * @param {StripePaymentResponse} paymentResponse    A stripe payment response
+ *                                                   object.
+ * @param {string}                paymentRequestType The payment request type
+ *                                                   used for payment.
+ *
+ * @return {Object} An object with the extra payment data.
+ */
+const getPaymentMethodData = ( paymentResponse, paymentRequestType ) => {
+	return {
+		payment_method: 'stripe',
+		stripe_source: paymentResponse.source
+			? paymentResponse.source.id
+			: null,
```

Comment:
Let's use optional chaining and null coalescing here
```suggestion
		stripe_source: paymentResponse.source?.id,
```

Reply:
Fixed in f13a35b04920d48c62ac4f806456c13aeff0cdc1